packages/fx-core/src/plugins/resource/bot/plugin.ts

```diff
@@ -779,51 +702,37 @@ export class TeamsBotImpl {
     Logger.info(Messages.SuccessfullyUpdatedBotMessageEndpoint);
   }
 
-  private async reuseExistingBotRegistration() {
-    const rawBotId = this.ctx!.answers![QuestionNames.GET_BOT_ID];
-    if (!rawBotId) {
-      throw new UserInputsError(QuestionNames.GET_BOT_ID, rawBotId as string);
-    }
-    const botId = rawBotId as string;
-
-    const rawBotPassword = this.ctx!.answers![QuestionNames.GET_BOT_PASSWORD];
-    if (!rawBotPassword) {
-      throw new UserInputsError(QuestionNames.GET_BOT_PASSWORD, rawBotPassword as string);
-    }
-    const botPassword = rawBotPassword as string;
-
-    return {
-      botId: botId,
-      botPassword: botPassword,
-    };
-  }
-
   private async createNewBotRegistrationOnAppStudio() {
     const appStudioToken = await this.ctx?.appStudioToken?.getAccessToken();
     CheckThrowSomethingMissing(ConfigNames.APPSTUDIO_TOKEN, appStudioToken);
     CheckThrowSomethingMissing(CommonStrings.SHORT_APP_NAME, this.ctx?.projectSettings?.appName);
 
-    if (
-      this.config.localDebug.botRegistrationCreated() &&
-      (await AppStudio.isAADAppExisting(appStudioToken!, this.config.localDebug.localObjectId!))
-    ) {
-      Logger.debug("Local bot has already been registered, just return.");
-      return;
-    }
-
     // 1. Create a new AAD App Registraion with client secret.
     const aadDisplayName = ResourceNameFactory.createCommonName(
       this.config.resourceNameSuffix,
       this.ctx?.projectSettings?.appName,
       MaxLengths.AAD_DISPLAY_NAME
     );
 
-    Logger.info(Messages.ProvisioningBotRegistration);
-    const botAuthCreds = await AADRegistration.registerAADAppAndGetSecretByAppStudio(
-      appStudioToken!,
-      aadDisplayName
-    );
-    Logger.info(Messages.SuccessfullyProvisionedBotRegistration);
+    let botAuthCreds: BotAuthCredential = new BotAuthCredential();
+    if (
+      this.config.localDebug.botAADCreated()
+      // if user input AAD, the object id is not required
+      // && (await AppStudio.isAADAppExisting(appStudioToken!, this.config.localDebug.localObjectId!))
+    ) {
+      botAuthCreds.clientId = this.config.localDebug.localBotId;
+      botAuthCreds.clientSecret = this.config.localDebug.localBotPassword;
+      botAuthCreds.objectId = this.config.localDebug.localObjectId;
+      Logger.debug(Messages.SuccessfullyGetExistingBotAadAppCredential);
+      return;
```

Comment:
Why we return here? In existing AAD scenario, it just means we don't need to create AAD for bot again, but we still need to do bot registration, right?